src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Infrastructure/KestrelTrace.cs

```diff
@@ -30,6 +30,9 @@ public class KestrelTrace : IKestrelTrace
         private static readonly Action<ILogger, string, Exception> _connectionDisconnect =
             LoggerMessage.Define<string>(LogLevel.Debug, 10, @"Connection id ""{ConnectionId}"" disconnecting.");
 
+        private static readonly Action<ILogger, string, Exception> _connectionRejected =
+            LoggerMessage.Define<string>(LogLevel.Warning, 11, @"Connection id ""{ConnectionId}"" rejected.");
```

Comment:
We decided when working on #1017 that if there's nothing the application developer could do to prevent a client from forcing the log (e.g. bad request), then the log level should not be higher than Information.

I think this is an exception to that rule though, because the client won't be getting a 503 (for now) and hitting the connection limit can cause connection refusals for clients that aren't responsible for the limit being hit.